internals/resource-managers/output.go

```diff
@@ -0,0 +1,161 @@
+package resourceprepares
+
+import (
+	"context"
+	"github.com/cisco-open/operator-tools/pkg/secret"
+	rcsv1alpha1 "github.com/dana-team/container-app-operator/api/v1alpha1"
+	rclient "github.com/dana-team/container-app-operator/internals/wrappers"
+	"github.com/go-logr/logr"
+	loggingv1beta1 "github.com/kube-logging/logging-operator/pkg/sdk/logging/api/v1beta1"
+	"github.com/kube-logging/logging-operator/pkg/sdk/logging/model/output"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type OutputManager struct {
+	Ctx       context.Context
+	K8sclient client.Client
+	Log       logr.Logger
+}
+
+const (
+	LogTypeElastic       string = "elastic"
+	LogTypeSplunk        string = "splunk"
+	KnativeConfiguration        = "serving.knative.dev/configuration"
+	NginxPraser                 = "nginx"
+	ElasticPort                 = 9200
+	SplunkHecPort               = 8088
+	ElasticSSLVersion           = "TLSv1_2"
+	BufferTimekey               = "1m"
+	BufferTimekeyWait           = "30s"
+	BufferTimekeyUseUtc         = true
+)
+
+// prepareResource prepares an output resource based on the provided capp.
+func (o OutputManager) prepareResource(capp rcsv1alpha1.Capp) loggingv1beta1.Output {
+	outputName := capp.GetName() + "-output"
+	protocol := "http"
+	if capp.Spec.LogSpec.SSLVerify {
+		protocol = "https"
+	}
+	if capp.Spec.LogSpec.Type == LogTypeElastic {
```

Comment:
I wont do an if  and create an object in this function. because every type will expand this function, ill make a map with string key type and every key will represent a log type the.  The value will be a function which creates the fitting output object